### PR TITLE
update doc and requirements.txt

### PR DESCRIPTION
**Pull request by AllentDan**

### CODE REVIEW

**Comment by lvhan028** [comment]
May rebase main to fix linting check

**Comment by lvhan028** [comment]
failed when installing lmdeploy under python3.10 
```
          raise VersionConflict(dist, req).with_context(dependent_req)
      pkg_resources.ContextualVersionConflict: (markdown-it-py 3.0.0 (/home/lvhan/projects/llmdeploy/.eggs/markdown_it_py-3.0.0-py3.10.egg), Requirement.parse('markdown-it-py<3.0.0,>=1.0.0'), {'mdit-py-plugins'})
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed
```

File: README.md [hunk 1]
```diff
@@ -48,6 +48,7 @@ conda create -n lmdeploy python=3.10
 conda activate lmdeploy
 git clone https://github.com/InternLM/lmdeploy.git
 cd lmdeploy
+pip install -r requirements.txt
```
Comment (lvhan028) [inline README.md:51]: `pip install  -e .` installs the dependent package automatically. We don't need to do it manually.

File: requirements.txt [hunk 2]
```diff
@@ -4,5 +4,9 @@ numpy
 pybind11
 safetensors
 setuptools
-transformers<4.30.0
-tritonclient==2.33.0
+protobuf
+gradio
+sentencepiece
+deepspeed # this is gonna be large
+transformers==4.30.2
```
[the remark this answers is not in the document]
Comment (lvhan028) [reply requirements.txt:9]: It is unnecessary to specify the transformers' version now. Besides, I found it brought package conflict between package markdown-it-py and mdit-py-plugins

**Comment by lvhan028** [comment]
When I run "python deploy.py ...", an error is raised:
```
    import torch
ModuleNotFoundError: No module named 'torch'
```

**Comment by lvhan028** [comment]
@APX103 could you help verify the quick start mentioned in the readme?

**Comment by APX103** [comment]
Verified, all tests passed @lvhan028

**Pull request merged**